pynxtools/dataconverter/template.py

```diff
@@ -125,6 +125,13 @@ def __contains__(self, k):
             k in self.required
         ])
 
+    def get(self, key: str, default=None):
+        """Proxies the get function to our internal __getitem__"""
+        try:
+            return self[key]
+        except KeyError:
+            return default
+
```

Comment:
No, Rubel. You got it wrong. It only calls get_optionality() when the if statements are not satisfied. When a key isn't found it does raise a KeyError. Have another look. Try a template.get("/key/doesn't/exist", "somekey") and you'll get the right results.

The last call to get_optionality is a way to conveniently access your inner dicts from template based on optionality. Template only has keys starting with "/".

Also, on slack you just mentioned that you want to keep the duplicate code you added instead of using this fix we did together on Wednesday.